Ajout Levnberg-Marquardt

OscarHerNA opened the pull request:
Merge d'un nouveau algotithme de minimisation dans la branche working(version correcte)

OscarHerNA committed:
Correction de CubicSpline et Hyman avec ajout dans Exemple de Test pour vérifier le résultat. Ajout de la possibilité de différente Spline comme Naturelle Clamped.

OscarHerNA committed:
Correction de CubicSpline et Hyman avec ajout dans Exemple de Test pour vérifier le résultat + Ajoujet exemple de différentes YieldCurve avec bump

OscarHerNA committed:
Ajout dans Euclid de LevenbergMarquardt et la calibration globale des instruments dans YieldCurve via BuilSwapCurveGlobal

OscarHerNA committed:
Ajout d un exemple pour la descente de gradient.

OscarHerNA committed:
Modification Courbe ESTR

OscarHerNA committed:
Correction Nom explicites

OscarHerNA committed:
Commentaires Levenberg

OscarHerNA committed:
Ajout Levenberg-Marquardt

OscarHerNA committed:
Modification de LevenbergMarquardt pour avoir la mthode simple et de J.Moré et laisser le choix de la difference finies et d'une maximisation . Implmentation de difference relative dans EndCriteria pour la convergence de la fonction. Modification de BuildSwapCurveGlobal optimise(gain de temps)

OscarHerNA committed:
Correction erreur signe pour  OptimizationType.Max dans la version OptimizeAdaptive()

OscarHerNA committed:
Ajout du jacobien et du gradient dans Differentiator et utilisation de cela dans levenberg-marquardt impliquant la modification de BuildSwapCurveGlobal

OscarHerNA committed:
Changement legers dans levenbergMarquardt

OscarHerNA committed:
version Fonctionnele avec while

OscarHerNA committed:
correction mineures

lhazoume committed:
interventions sur le code

OscarHerNA committed:
Version modifie de OptimizeAdaptive

OscarHerNA committed:
Update Levenberg + Ajout d'un test dans Exemple

OscarHerNA committed:
Suppression du choix de Min et Max dans LevenbergMarquardt.

OscarHerNA committed:
Version Avant Modification

OscarHerNA committed:
Correction erreur signe

OscarHerNA committed:
suppression champs inutiles

OscarHerNA committed:
Update LevenbergMarquardt.cs

lhazoume reviewed (changes requested):
Mise à jour du ShouldStop dans le EndCriteria
Repenser le Change Epsilon
Enlever les commentaires dans la descente du gradient
Renommer _bcType en _boundaryType
Supprimer la previousValue dans EndCriteria et utiliser la List<double> _history

OscarHerNA committed:
Corrige les points soulevés dans la revue : Mise à jour du ShouldStop dans le EndCriteria
Repenser le Change Epsilon
Enlever les commentaires dans la descente du gradient
Renommer _bcType en _boundaryType
Supprimer la previousValue dans EndCriteria et utiliser la List _history et reorganisation) des interpolateurs dans Palma

OscarHerNA committed:
Merge branch 'Optimizers-Levenberg-Marquardt' of https://github.com/lhazoume/Euclid into Optimizers-Levenberg-Marquardt

OscarHerNA committed:
Correction var interpolateur Brick

OscarHerNA committed:
xml manquant

OscarHerNA committed:
Modifications mineures

OscarHerNA committed:
Changement mineures

OscarHerNA committed:
Gestion lambdaOutput et PenltyFactor Output dans levenberg

OscarHerNA committed:
Changement mineures

OscarHerNA committed:
changements tres mineures

OscarHerNA committed:
Changement du need refresh dans pour que les interpolateur global fonctionn avec BuildSwapCurve +changement mineures

lhazoume reviewed (approved):
Excellent travail
